src/main/java/com/fincons/taskmanager/controller/AttachmentController.java

```diff
@@ -0,0 +1,184 @@
+package com.fincons.taskmanager.controller;
+
+import com.fincons.taskmanager.dto.AttachmentDTO;
+import com.fincons.taskmanager.entity.Attachment;
+import com.fincons.taskmanager.exception.DuplicateException;
+import com.fincons.taskmanager.exception.ResourceNotFoundException;
+import com.fincons.taskmanager.mapper.AttachmentMapper;
+import com.fincons.taskmanager.service.attachmentService.AttachmentService;
+import com.fincons.taskmanager.utility.GenericResponse;
+import com.fincons.taskmanager.utility.ValidateFields;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@CrossOrigin("*")
+@RequestMapping("/task-manager")
+public class AttachmentController {
+
+    @Autowired
+    private AttachmentService attachmentService;
+    @Autowired
+    private AttachmentMapper modelMapperAttachment;
+
+    @GetMapping(value = "${attachment.find-by-code}")
+    public ResponseEntity<GenericResponse<AttachmentDTO>> getAttachmentByCode(@RequestParam String code) {
+        try {
+            ValidateFields.validateSingleField(code);
+            Attachment attachment = attachmentService.getAttachmentByCode(code);
+            AttachmentDTO attachmentDTO = modelMapperAttachment.mapToDTO(attachment);
+            GenericResponse<AttachmentDTO> response = GenericResponse.success(
+                    attachmentDTO,
+                    "Success: Found Attachment with CODE " + code + ".",
+                    HttpStatus.OK
+            );
+            return ResponseEntity.ok(response);
+        } catch (IllegalArgumentException iae) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
+                    GenericResponse.error(
+                            iae.getMessage(),
+                            HttpStatus.BAD_REQUEST
+                    )
+            );
+        } catch (ResourceNotFoundException rnfe) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(
+                    GenericResponse.error(
+                            rnfe.getMessage(),
+                            HttpStatus.NOT_FOUND
+                    )
+            );
+        }
+    }
+
+    @GetMapping(value = "${attachment.list}")
+    public ResponseEntity<GenericResponse<List<AttachmentDTO>>> getAllAttachment() {
```

Comment:
get all attachement should be plural name

Reply:
Resolve